### PR TITLE
feat: Atlas databuilder compatibility

### PR DESCRIPTION
**Pull request by mgorsk1**
### Summary of Changes

Introduces support between Atlas and majority of databuilder models.

The change was done with minimal amount of changes in the proxy itself, it's backwards compatible and unlocks both push and pull mode for ingesting data to Atlas:
- **push** using Hive Hook and Spark Atlas Connector to collect hive table & lineage metadata
- **pull** using Amundsen Databuilder to leverage available extractors 

#### New compatible models
- Application
- Badge
- Custer
- Column
- Column Usage
- Dashboard Usage
- Dashboard Owner
- Database
- Schema
- Source
- Table
- Table Lineage
- Table Owner
- Table Usage
- Tag
- User
- Watermarks

#### Already supported models
- Dashboard Chart
- Dashboard Execution
- Dashboard Last Modified
- Dashboard Metadata
- Dashboard Query
- Dashboard Table

#### Two models were temporarily excluded from this effort as they might require redesign in metadata AtlasProxy
- Column Statistics
- Programmatic Descriptions

#### Corresponding PR for introducing new Atlas Types

https://github.com/dwarszawski/amundsen-atlas-types/pull/35

### Tests

tbd

### Documentation

- updated

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by mgorsk1** [comment]
cc @dechoma

**Comment by mgorsk1** [comment]
#keepalive

File: databuilder/databuilder/extractor/atlas_search_data_extractor.py [hunk 1]
```diff
@@ -86,6 +86,43 @@ def get_chart_names(queries: Optional[List]) -> List[str]:
 
         return AtlasSearchDataExtractorHelpers.get_display_text(charts)
 
+    @staticmethod
+    def get_table_database(qualified_name: str) -> str:
+        try:
+            result = AtlasTableKey(qualified_name).get_details()['database']
+        except Exception:
```
[the remark this answers is not in the document]
Comment (mgorsk1) [reply databuilder/databuilder/extractor/atlas_search_data_extractor.py:93]: sure, I simplified it even more

File: databuilder/databuilder/extractor/atlas_search_data_extractor.py [hunk 2]
```diff
@@ -86,6 +86,43 @@ def get_chart_names(queries: Optional[List]) -> List[str]:
 
         return AtlasSearchDataExtractorHelpers.get_display_text(charts)
 
+    @staticmethod
+    def get_table_database(qualified_name: str) -> str:
```
[the remark this answers is not in the document]
Comment (mgorsk1) [reply databuilder/databuilder/extractor/atlas_search_data_extractor.py:90]: makes sense!

File: databuilder/databuilder/extractor/atlas_search_data_extractor.py [hunk 3]
```diff
@@ -86,6 +86,43 @@ def get_chart_names(queries: Optional[List]) -> List[str]:
...
+        unique = 0
+
+        for reader in readers:
+            reader_status = reader.get('status')
+            entity_status = reader.get('relationshipAttributes', dict()).get('entity', dict()).get('entityStatus', '')
+            relationship_status = reader.get('relationshipAttributes',
+                                             dict()).get('entity',
+                                                         dict()).get('relationshipStatus', '')
+
+            if reader_status == 'ACTIVE' and entity_status == 'ACTIVE' and relationship_status == 'ACTIVE':
```
[the remark this answers is not in the document]
Comment (mgorsk1) [reply databuilder/databuilder/extractor/atlas_search_data_extractor.py:118]: true

**Review by dechoma — APPROVED** [review]

**Pull request merged**